_posts/2023-17-10-remote-attestation-blog-part1.markdown

```diff
@@ -0,0 +1,137 @@
+
+For measured boot attestation to be fully useful, Keylime operators should provide their own measured boot reference state and measured boot policy. Ideally, you want to create a policy that is both meaningful and generic enough to be applied to a set of nodes. One can change the pre-provided mb_policy to suit their needs rather than attempting to write a brand new one from scratch.
+
+If you want to read more about how to enable measured boot on Keylime, you can do that [here](https://keylime.readthedocs.io/en/latest/user_guide/use_measured_boot.html). 
+
+<h4 style="text-align: center; color:#99A3A4 ;"> Keylime, IMA, and Runtime Integrity Monitoring </h4>
+
+To use Keylime’s Runtime Integrity Monitoring, there are a few steps that you will need to take to either enable Linux IMA or ensure that it is already running. As we learned a few sections ago, IMA is used to measure the contents of files at runtime. However, before it can do that, you will have to tell it what to measure using an ***ima-policy***. In the Keylime repository, you will find an example policy that can be used, and in the documentation, you will find instructions on enabling IMA. 
+
+In Keylime, a ***runtime policy*** is a list of “golden” cryptographic hashes of files in their untampered state that are used for IMA verification. The Keylime verifier will load the runtime policy and use it to validate the agent’s file states that are loaded onto the TPM. If the files have been tampered with, the hashes will not match, and Keylime will place the agent into a failed state. The Keylime agent and operating system itself are not deemed trustworthy by default. Given the setup only after the successful initial attestation is the system deemed trustworthy, but it still can leave the trusted state at any moment and is, therefore, continuously attested.
+
+If you want to read more about how to build a runtime policy for Keylime and then deploy it, you can do that [here](https://keylime.readthedocs.io/en/latest/user_guide/runtime_ima.html). 
+
+<h4 style="text-align: center; color:#99A3A4 ;"> How Keylime Can Help You </h4>
+
+Remote attestation is a way for you to create peace of mind that you can and will detect when something is amiss inside a cluster. By building a chain of trust from hardware to software, you can ensure your data is protected. 
+
+Keylime is commonly used either on bare metal hardware or in VMs where the TPM is emulated but, from the VM side, treated the same as a hardware TPM. In this article, we have seen what Keylime can detect:
+
+<ul>
+    <li> Modification of the boot process </li>
+    <li> Runtime file and system integrity </li>
+</ul>
+
+While it won’t stop anything after an alert is raised, Keylime is still a great tool to have because it will tell you what has happened and where. Perfect for someone managing a lot of machines!
```

Comment:
```
While it won’t stop anything after an alert is raised,
```

You can implement what ever you want around the revocation hook that Keylime provides, e.g. remove the node from the cluster when it Keylime detects that it is compromised.